google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSource.java

```diff
@@ -127,17 +115,17 @@ public File getFile() {
         new File(mavenProject.getBuildDirectory()).getPath() + File.separator
             + mavenProject.getFinalName();
 
-    XmlFile appEngineWebXml = AppEngineAssetProvider.getInstance()
-        .loadAppEngineStandardWebXml(project, Collections.singletonList(getModule()));
     AppEngineProjectService projectService = AppEngineProjectService.getInstance();
 
     // The environment will be null for newly deserialized deployment sources to ensure freshness.
     // In this case, we need to reload the environment.
     if (environment == null) {
-      environment = projectService.getModuleAppEngineEnvironment(appEngineWebXml);
+      environment = projectService.getModuleAppEngineEnvironment(getModule()).orElseThrow(
+          () -> new RuntimeException("No environment.")
```

Comment:
I just performed that test and nothing breaks. When the run config is reopened, the deployables list only contains the user specified one.

By now having to unwrap Optionals, we run into problems like this in other places too. Let me know if you have a better suggestion than to throw the exception here.

Off the top of my head, it would have to involve going back to using nulls (and risking NPEs later down the road) or creating an UNKNOWN environment.